x-pack/plugins/rule_registry/server/event_log/log/event_log_bootstrapper.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { inspect } from 'util';
+import { Logger } from 'kibana/server';
+
+import { IIndexBootstrapper, IndexSpecification } from '../elasticsearch';
+import { ReadySignal, createReadySignal } from '../utils/ready_signal';
+
+interface ConstructorParams {
+  indexSpec: IndexSpecification;
+  indexBootstrapper: IIndexBootstrapper;
+  isWriteEnabled: boolean;
+  logger: Logger;
+}
+
+export class EventLogBootstrapper {
+  private readonly indexSpec: IndexSpecification;
+  private readonly indexBootstrapper: IIndexBootstrapper;
+  private readonly isWriteEnabled: boolean;
+  private readonly logger: Logger;
+  private readonly bootstrappingFinished: ReadySignal<EventLogBootstrappingResult>;
+
+  constructor(params: ConstructorParams) {
+    this.indexSpec = params.indexSpec;
+    this.indexBootstrapper = params.indexBootstrapper;
+    this.isWriteEnabled = params.isWriteEnabled;
+    this.logger = params.logger.get('EventLogBootstrapper');
+    this.bootstrappingFinished = createReadySignal<EventLogBootstrappingResult>();
+  }
+
+  public start(): void {
```

Comment:
I have a couple of concerns about the interface of the `EventLogBootstrapper`. Currently it is used like this:

```js
this.logBootstrapper.start();
const { success } = await this.logBootstrapper.waitUntilFinished();
if (!success) {
     throw new Error(`Event log bootstrapping failed, logName="${logName}"`);
}
```

- What is the reason for the `start()` method to be sync? It is asynchronous under the hood and used in conjunction with `waitUntilFinished()`. IMO the relationship between these two methods is quite easy to overlook from the developer's perspective (e.g., what will happen if we call `start()` and forget to call `await waitUntilFinished()`). 
- `waitUntilFinished()` returns boolean value: `true` in case of success and `false` otherwise. It could become a problem when we decide to introduce error handling. I think having the original error instead of `false` could be helpful in various scenarios, e.g., for retry logic in case of 503 or showing callouts in case of missing privileges.

Taking into account the above, I propose to change the `EventLogBootstrapper` interface, so we can use like that:

```js
try {
  await this.logBootstrapper.start();
} catch (e) {
  if (e.status_code === 403) {
    // Handle insuffincient privileges
  } else if (e.status_code === 503) {
    // Retry
  } else {
    throw new Error(`Event log bootstrapping failed, logName="${logName}"`);
  }
}
```

Reply:
> What is the reason for the `start()` method to be sync?

It's a leftover. This API was useful when index bootstrapping was being done on the plugin start, and `start()` was called by the plugin itself, but `waitUntilFinished()` was supposed to be called by the client code when trying to resolve a log. Now this separation is likely not needed.

Error handling is a valid point. But the thing is: this implementation of index bootstrapping is based on the `event_log` plugin implementation; in `security_solution` there's a (slightly) different one; https://github.com/elastic/kibana/pull/98935 introduced yet another way of bootstrapping. So we need to combine/converge all approaches into a single one first, and after that we'll be able to add proper error handling.

So your concerns are valid, and I like the suggestions, but I'd suggest to address robust index bootstrapping in a separate PR. What I can easily do now is merge `start()` and `waitUntilFinished()` into a single method.

Reply:
Sounds good 👍 
Let's merge `start()` and `waitUntilFinished()` as for now. I think we can also remove `ReadySignal` then.

Reply:
Addressed